python/activator/middleware_interface.py

```diff
@@ -56,25 +58,46 @@ class MiddlewareInterface:
         Path to a butler repo containing the calibration and other data needed
         for processing images as they are received.
     image_bucket : `str`
-        Google storage bucket where images will be written to as they arrive.
+        Storage bucket where images will be written to as they arrive.
+        See also ``prefix``.
     instrument : `str`
-        Name of the instrument taking the data, for populating butler
-        collections and dataIds.
+        Full class name of the instrument taking the data, for populating
+        butler collections and dataIds. Example: "lsst.obs.lsst.LsstCam"
+    prefix : `str`, optional
+        URI identification prefix; prepended to ``image_bucket`` when
+        constructing URIs to retrieve incoming files.
+    tmpdir_suffix : `str`, optional
+        Suffix of local temporary butler; if None, use the current process id.
     """
-    def __init__(self, input_repo: str, image_bucket: str, instrument: str):
-
+    def __init__(self, input_repo: str, image_bucket: str, instrument: str,
+                 prefix: str = "gs://", tmpdir_suffix=None):
+        self.prefix = prefix
         # self.src = Butler(input_repo, writeable=False)
         _log.info(f"Butler({input_repo}, writeable=False)")
         self.image_bucket = image_bucket
-        self.instrument = instrument
+        self.instrument = lsst.obs.base.utils.getInstrument(instrument)
 
-        self.repo = f"/tmp/butler-{os.getpid()}"
+        if tmpdir_suffix is None:
+            tmpdir_suffix = f"{os.getpid()}"
+        self.repo = f"/tmp/butler-{tmpdir_suffix}"
         if not os.path.exists(self.repo):
             _log.info(f"Making local Butler {self.repo}")
             Butler.makeRepo(self.repo)
         else:
             _log.info(f"Local Butler {self.repo} exists")
-        self.dest = Butler(self.repo, writeable=True)
+        butler = Butler(self.repo, writeable=True)
+        self.instrument.register(butler.registry)
```

Comment:
Good question. I don't know whether we actually need to start with a partial butler, or create one programmatically like this. This is certainly at least a stopgap, but maybe we want to actually built up the local butler from scratch like this, so that it's always fully up-to-date with the running butler schema, etc? If we build the local butler entirely from export/import, we'll have to update those exports occasionally.

Reply:
We'd be exporting from the "nightly" calib repo. Wouldn't we need to keep it up to date anyway for the import to succeed?

I don't know what you mean by "partial Butler", but I know from `ap_verify` that creating an empty repository and then immediately importing is sufficient to create a repository that's in a usable state.

Reply:
Yes, by "partial butler" I meant starting from an import statement.

I'm leaning towards leaving this line here as is, with a TODO about needing to import a "base" repo in the future. I think you're right that import is the way we want to go eventually, but until we have an example of what that import file will look like (and one I can check-in for the tests here), I need this to be able to do ingest tests with fake data.